### DISCUSSION TITLE
Need to learn JS

### DISCUSSION THREAD

**Discussion by deepaknewjs** [community/community#138939] [Programming Help]
### Select Topic Area

Question

### Body

Like if I learn about the functions, loop, array all basic things aboutJS, then how to use those and where. And how to think what will come here in perticual section in JS. Like how to think where and what code do I need to do this perticular thing.

**Comment by jovanovic-djo** [comment]
There is inevitable part of learning path, that is tutorial hell, i would suggest you to start with any tutorial course that you find acceptable, just to figure out basics.

Be consistent.
Don't give up.
Have fun.

When you figure out basics, I would suggest you to start working on small projects (choice by your liking) and commit them to your github acc.

That step is most important, because you no longer rely on tutorials, you face problem solving on your own through googling, looking on stackoverflow or even using chatgpt.

@deepaknewjs  Enjoy your journey.

**Reply by deepaknewjs** [reply]
Thank you so much for your advice. I will keep it in mind

**Reply by jovanovic-djo** [reply]
If you have additional inquiries feel free to ask.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @deepaknewjs ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by joemar25** [comment]
@deepaknewjs when you're learning JavaScript, it can be a little confusing at first to know where to apply things like functions, loops, and arrays. The best way to start is by practicing through online courses or tutorials that give you hands-on coding experience. For example, when you use a loop, you're probably dealing with repetitive tasks like going through a list of items (an array). Functions are useful when you want to organize code into reusable blocks that perform specific tasks, so you don't repeat yourself.

As you work through different challenges, try to break things down. Ask yourself: 'What am I trying to achieve here?' If you're building something like a calculator, you might use functions for each operation (add, subtract, multiply, divide) and a loop to process user input. Keeping a journal or writing down your code steps helps keep your mind active and makes learning easier.

The more you code, the better you'll understand where to apply different concepts. So just keep practicing and experimenting!

**Reply by deepaknewjs** [reply]
Thank you so much for your advice ☺️. The section you mentioned in what to use where is really helpful for me. It is the main problem for me, if you have any other advice in same please feel free to add. Thank you in advance 🙂

**Reply by joemar25** [reply]
😄 That's good to hear.. @deepaknewjs, one fun idea is to dive into building small projects that excite you, like a simple game 🎮 or a to-do 📝 list app. It's a great way to see how functions, loops, and arrays all come together in real situations. And yeh, designing a mini-project from scratch can be tough, but it's super rewarding and a fantastic learning experience! also, while AItools can be handy for staying focused and getting quick tips, try not to over-rely on them when creating your projects... the real magic happens when you tackle challenges head-on and figure things out yourself. 😇

**Comment by cucer** [comment]
One of my favorite methods to learn new coding language is very simple.

- learn fundamentals
- use these fundamentals on basic CRUD projects
- use these CRUD skills on both relational db and no-sql db
- learn how to deploy your projects on google cloud and aws

I hope it helps...

**Comment by Xelliese** [comment]
Once you get the basics down like loops, arrays, and functions, it’s really about practice and building stuff. Start with small projects, like making a to-do list or a simple calculator, and you’ll start seeing patterns. As for "what code goes where," that comes with time and trial/error.

**Comment by ravin00** [comment]
Try freecodecamp and practice the basics. Practice is what would i recommend.

**Comment by iremozdemr** [comment]
If you want a very detailed tutorial, you can try this: [https://javascript.info](https://javascript.info)
